files/scripts/supervisor-proc-exit-listener

```diff
@@ -52,24 +52,25 @@ def main(argv):
             processname = payload_headers['processname']
             groupname = payload_headers['groupname']
 
-            config_db = swsssdk.ConfigDBConnector()
-            config_db.connect()
-            container_features_table = config_db.get_table(CONTAINER_FEATURE_TABLE_NAME)
-            if not container_features_table:
-                syslog.syslog(syslog.LOG_ERR, "Unable to retrieve container features table from Config DB. Exiting...")
-                sys.exit(2)
-
-            if not container_features_table.has_key(container_name):
-                syslog.syslog(syslog.LOG_ERR, "Unable to retrieve features for container '{}'. Exiting...".format(container_name))
-                sys.exit(3)
-
-            restart_feature = container_features_table[container_name].get('auto_restart')
-            if not restart_feature:
-                syslog.syslog(syslog.LOG_ERR, "Unable to determine auto-restart feature status for container '{}'. Exiting...".format(container_name))
-                sys.exit(4)
+            if container_name != 'database':
+                config_db = swsssdk.ConfigDBConnector()
+                config_db.connect()
+                container_features_table = config_db.get_table(CONTAINER_FEATURE_TABLE_NAME)
+                if not container_features_table:
+                    syslog.syslog(syslog.LOG_ERR, "Unable to retrieve container features table from Config DB. Exiting...")
+                    sys.exit(2)
+
+                if not container_features_table.has_key(container_name):
+                    syslog.syslog(syslog.LOG_ERR, "Unable to retrieve features for container '{}'. Exiting...".format(container_name))
+                    sys.exit(3)
+
+                restart_feature = container_features_table[container_name].get('auto_restart')
+                if not restart_feature:
+                    syslog.syslog(syslog.LOG_ERR, "Unable to determine auto-restart feature status for container '{}'. Exiting...".format(container_name))
+                    sys.exit(4)
 
             # If auto-restart feature is enabled and a critical process exited unexpectedly, terminate supervisor
-            if restart_feature == 'enabled' and expected == 0 and (processname in critical_processes or groupname in critical_processes):
+            if (container_name == 'database' or restart_feature == 'enabled') and expected == 0 and (processname in critical_processes or groupname in critical_processes):
```

Comment:
This line has gotten long and complex. Can you either break it into two lines or two conditionals?

Reply:
Will fix it.